src/realm/sync/noinst/client_reset.cpp

```diff
@@ -272,23 +352,19 @@ void client_reset::transfer_group(const Transaction& group_src, Transaction& gro
 
     // Create new tables in dst if needed.
     for (auto table_key : group_src.get_table_keys()) {
+        if (!group_src.table_is_public(table_key))
+            continue;
         ConstTableRef table_src = group_src.get_table(table_key);
         StringData table_name = table_src->get_name();
-        if (!table_name.begins_with("class"))
-            continue;
         bool has_pk = sync::table_has_primary_key(*table_src);
         TableRef table_dst = group_dst.get_table(table_name);
         if (!table_dst) {
             // Create the table.
-            if (!has_pk) {
-                sync::create_table(group_dst, table_name);
-            }
-            else {
-                auto pk_col_src = table_src->get_primary_key_column();
-                DataType pk_type = DataType(pk_col_src.get_type());
-                StringData pk_col_name = table_src->get_column_name(pk_col_src);
-                group_dst.add_table_with_primary_key(table_name, pk_type, pk_col_name, pk_col_src.is_nullable());
-            }
+            REALM_ASSERT(has_pk); // a sync table will have a pk
```

Comment:
It is meant to exclude internal tables such as object-store's "metadata" table. To be honest, I hadn't thought about embedded objects and I expect they won't work because they do not have primary keys. I will have to add support for these.